src/libraries/System.Security.Cryptography.Algorithms/ref/System.Security.Cryptography.Algorithms.cs

```diff
@@ -15,6 +15,9 @@ public abstract partial class Aes : System.Security.Cryptography.SymmetricAlgori
         public static new System.Security.Cryptography.Aes? Create(string algorithmName) { throw null; }
     }
     [System.Runtime.Versioning.UnsupportedOSPlatformAttribute("browser")]
+    [System.Runtime.Versioning.UnsupportedOSPlatformAttribute("ios")]
+    [System.Runtime.Versioning.UnsupportedOSPlatformAttribute("tvos")]
+    [System.Runtime.Versioning.UnsupportedOSPlatformAttribute("maccatalyst")]
```

Comment:
I think we generally omit `maccatalyst` when there is already `ios` exclusion

/cc @steveisok

Reply:
I was under the impression that they just were not added yet. They are certainly present on same places and not on others.

I'm bit worried about any potential asymmetry of the APIs. For example, it's fine to build the libraries with `net6.0-ios` target and it gets consumed by Mac Catalyst through fallback. However, if I do some runtime check with `OperatingSystem.IsIOS()` I also have to explicitly test for `IsMacCatalyst` (and `IsTvOS`) because there's no fallback there. I mentally map the `UnsupportedOSPlatform` declarations to the `OperatingSystem` APIs (which is how the platform analyzer maps the code branches) so I expect there to be symmetry of both of requiring explicit Mac Catalyst declaration/check.

Reply:
@MaximLipnin is looking at how the analyzer behaves for Catalyst because I'm not totally sure if we will get the same thing as ios.

Reply:
As far as I can tell the [Platform Compatibility Analyzer](https://github.com/dotnet/roslyn-analyzers/tree/main/src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/InteropServices) has no special case for Mac Catalyst. Hence it maintains the symmetry between `OperatingSystem.IsXXX` APIs and `UnsupportedOSPlatform("xxx")`. `OperatingSystem.IsIOS()` returns `false` on Mac Catalyst.

Reply:
Based on what I've got from building the net6-mobile-samples using the latest SDK, the platform compatibility analyzer doesn't consider iOS as a superset for MacCatalyst, so it makes sense to use a separate `UnsupportedOSPlatform("maccatalyst")]` annotation.

Reply:
Thanks for double checking my assumption!